master/buildbot/status/web/logs.py

```diff
@@ -212,11 +212,14 @@ def getChild(self, path, req):
                 if log.hasContents():
                     content = log.getText()
                     if isinstance(log, HTMLLogFile):
-                        if log.content_type == 'json':
-                            return JSONTestResource(log, self.step_status)
-                        elif log.content_type == 'xml' or ('xml-stylesheet' in content or 'nosetests' in content):
+                        if hasattr(log, 'content_type'):
```

Comment:
I think In this case we need to upgrade the serialized object to handle content_type so we don't depend on this check